narwhals/_arrow/dataframe.py

```diff
@@ -285,8 +286,18 @@ def join(
             ),
         )
 
-    def drop(self, *columns: str) -> Self:
-        return self._from_native_dataframe(self._native_dataframe.drop(list(columns)))
+    def drop(self: Self, *columns: str, strict: bool = True) -> Self:
```

Comment:
Ok hear me out, I know that we don't add defaults in here, but we _sometimes_ used compliant dataframe drop method (e.g. in `join`). I can add it in the call there